### PR TITLE
README

### PR DESCRIPTION
**Pull request by julien-c**
(also pushing this first version to `main` to be able to bundle on PyPI, but opening this PR for comments/review)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,3 +11,93 @@
...
+huggingface-cli login
+huggingface-cli logout
+huggingface-cli whoami
+huggingface-cli repo create
+```
+
+### Need to upload large (>5GB) files?
+
+To upload large files (>5GB 🔥), you need to install the custom transfer agent for git-lfs, bundled in this package. Spec for LFS custom transfer agent is:
+https://github.com/git-lfs/git-lfs/blob/master/docs/custom-transfers.md
```
Comment (sgugger) [inline README.md:90]:
```suggestion
To upload large files (>5GB 🔥), you need to install the custom transfer agent for git-lfs, bundled in this package. You can learn how to add an LFS custom transfer agent [here](https://github.com/git-lfs/git-lfs/blob/master/docs/custom-transfers.md).
```

File: README.md [hunk 2]
```diff
@@ -11,3 +11,93 @@
...
+
+To install, just run:
+
+```bash
+$ huggingface-cli lfs-enable-largefiles
+```
+
+This should be executed once for each model repo that contains a model file >5GB. It's documented in the error
+message you get if you just try to git push a 5GB file without enabling it before.
```
Comment (sgugger) [inline README.md:99]:
```suggestion
This should be executed once for each model repo that contains a model file >5GB. If you just try to push a file bigger than 5GB without running that command, you will get an error with a message reminding you to run it.
```
Not too happy about my tweaking either, so feel free to improve the wording.

File: README.md [hunk 3]
```diff
@@ -11,3 +11,93 @@
...
+
+Finally, there's a `huggingface-cli lfs-multipart-upload` command but that one is internal (called by lfs directly) and is not meant to be called by the user.
```
Comment (sgugger) [inline README.md:101]: Is it really worth mentioning since we don't even tell the user what it does?

**Review by sgugger — APPROVED** [review]
Exciting!

File: README.md (hunk 3 above)
Comment (julien-c) [reply README.md:101]: probably not, I'll remove

File: README.md (hunk 2 above)
Comment (julien-c) [reply README.md:99]: i think yours is better

File: README.md [hunk 4]
```diff
@@ -11,3 +11,91 @@
...
+```bash
+huggingface-cli login
+huggingface-cli logout
+huggingface-cli whoami
+huggingface-cli repo create
+```
+
+### Need to upload large (>5GB) files?
+
+To upload large files (>5GB 🔥), you need to install the custom transfer agent for git-lfs, bundled in this package. You can learn how to add an LFS custom transfer agent [here](https://github.com/git-lfs/git-lfs/blob/master/docs/custom-transfers.md).
```
Comment (julien-c) [inline README.md:89]:
```suggestion
To upload large files (>5GB 🔥), you need to install the custom transfer agent for git-lfs, bundled in this package. 
```

I think I'll just skip the custom transfer agent spec too (might not be super helpful to most users)

**Review by patrickvonplaten — APPROVED** [review]
Great!

File: README.md [hunk 5]
```diff
@@ -11,3 +11,91 @@
...
+Check out the source code for all possible params (we'll create a real doc page in the future).
+
+<br>
+
+## Publish models to the huggingface.co hub
+
+Uploading a model to the hub is super simple too:
+- create a model repo directly from the website, at huggingface.co/new (models can be public or private, and are namespaced under either a user or an organization)
+- clone it with git
+- install [lfs](https://git-lfs.github.com/) with `git lfs install` if you haven't done that before
```
Comment (Pierrci) [inline README.md:65]:
```suggestion
- [download and install git lfs](https://git-lfs.github.com/) if you don't already have it on your machine (you can check by running a simple `git lfs`)
```
`git lfs install` is more the setup, you need to download it first

**Review by Pierrci — APPROVED** [review]

**Review by thomwolf — APPROVED** [review]
🥰

**Pull request merged**